src/Native/System.Native/pal_process.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "../config.h"
+#include "pal_process.h"
+#include <sys/resource.h>
+#include <assert.h>
+
+// Each platform type has it's own RLIMIT values but the same name, so we need
+// to convert our standard types into the platform specific ones.
+int32_t ConvertRlimitResourcesToPlatformValue(RlimitResources value)
+{
+    switch (value)
+    {
+        case RlimitResources::PAL_RLIMIT_CPU:
+            return RLIMIT_CPU;
+        case RlimitResources::PAL_RLIMIT_FSIZE:
+            return RLIMIT_FSIZE;
+        case RlimitResources::PAL_RLIMIT_DATA:
+            return RLIMIT_DATA;
+        case RlimitResources::PAL_RLIMIT_STACK:
+            return RLIMIT_STACK;
+        case RlimitResources::PAL_RLIMIT_CORE:
+            return RLIMIT_CORE;
+        case RlimitResources::PAL_RLIMIT_AS:
+            return RLIMIT_AS;
+        case RlimitResources::PAL_RLIMIT_RSS:
+            return RLIMIT_RSS;
+        case RlimitResources::PAL_RLIMIT_MEMLOCK:
+            return RLIMIT_MEMLOCK;
+        case RlimitResources::PAL_RLIMIT_NPROC:
+            return RLIMIT_NPROC;
+        case RlimitResources::PAL_RLIMIT_NOFILE:
+            return RLIMIT_NOFILE;
+        default:
+            assert(!"Unknown RLIMIT value");
```

Comment:
A trick is to put this assert after the switch instead of default, then clang will warn if not all the RlimitResources enum values are handled. See errno shim for example.